src/plugins/intel_gpu/include/intel_gpu/runtime/layout.hpp

```diff
@@ -288,15 +288,29 @@ struct padding {
     /// @return Tensor with padding for bottom/right/upper bounds of data.
     tensor upper_size() const { return _upper_size; }
 
+    void set_dynamic_pad(tensor dynamic_pad_dims) {
+        _dynamic_pad_dims = dynamic_pad_dims;
+    }
+
+    tensor get_dynamic_pad_dims() const {
+        return _dynamic_pad_dims;
+    }
     /// @brief
     /// @param lower_sizes Top-left padding sizes. See @ref tensor::tensor(const std::vector<value_type>&, value_type) for details.
     /// @param upper_sizes Bottom-right padding sizes. See @ref tensor::tensor(const std::vector<value_type>&, value_type) for details.
     /// @param filling_value Filling value for padding area.
-    padding(const std::vector<tensor::value_type>& lower_sizes, const std::vector<tensor::value_type>& upper_sizes, float filling_value = 0.0f)
-        : _lower_size(to_abs(lower_sizes), 0), _upper_size(to_abs(upper_sizes), 0), _filling_value(filling_value) {}
+    padding(const std::vector<tensor::value_type>& lower_sizes,
+            const std::vector<tensor::value_type>& upper_sizes,
+            float filling_value = 0.0f,
+            tensor dynamic_pad_dims = tensor(0))
```

Comment:
Maybe we can use ov::PartialShape as well? I'd want to remove cldnn::tensor in the end. Or maybe we can make lower/upper pad ov::PartialShape and remove this extra parameter at all?

Reply:
The latter sounds good.

Reply:
But it seems quite a work, so let me separate the refactoring task. I'll create corresponding ticket.